src/oop/static_modifier/hard/q2/MainTest.java

```diff
@@ -0,0 +1,162 @@
+        assertTrue(testClass.hasModifier(staticAttributeName, "static"), String.format(wrongModifierMessage, staticAttributeName, className));
+    }
+
+    @Test
+    public void lightbulbHasRequiredConstructor() {
+        String missingConstructorMessage = "The %s class is missing a required constructor. Make sure that it is named correctly and has the correct parameters.";
+        String wrongAccessModifier = "The %s class constructor has the wrong visibility modifier. Make sure that it is visible from all other classes.";
+        Class<?>[] constructorArgs = new Class[]{String.class};
+        assertTrue(testClass.hasConstructor(constructorArgs), String.format(missingConstructorMessage, className));
+        assertTrue(testClass.hasModifier(constructorArgs, "public"), String.format(wrongAccessModifier, className));
+    }
+
+    private static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of("Incandescent", 100),
+                Arguments.of("Halogen", 200),
+                Arguments.of("Compact Fluorescent Lamp (CFL)", 110),
+                Arguments.of("LED", 20),
+                Arguments.of("Tungsten Filament", 130),
+                Arguments.of("High-Intensity Discharge (HID)", 40),
+                Arguments.of("Carbon Filament", 20),
+                Arguments.of("Fluorescent", 300),
+                Arguments.of("Full Spectrum", 70),
+                Arguments.of("Edison Bulb", 40)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void lightbulbConstructorInitializesValuesCorrectly(String value1) throws Throwable {
+        String missingConstructorMessage = "The %s class is missing a required constructor. Make sure that it is named correctly and has the correct parameters.";
+        String wrongAccessModifier = "The %s class constructor has the wrong visibility modifier. Make sure that it is visible from all other classes.";
+        Class<?>[] constructorArgsDef = new Class[]{String.class};
+        assertTrue(testClass.hasConstructor(constructorArgsDef), String.format(missingConstructorMessage, className));
+        assertTrue(testClass.hasModifier(constructorArgsDef, "public"), String.format(wrongAccessModifier, className));
+        String wrongValueMessage = "The %s constructor did not initialize the %s attribute to the correct value based on the parameters passed to the constructor.";
+        Object[][] constructorArgs = {
+                {value1, String.class}
+        };
+        Object checkupInstance = testClass.createInstance(constructorArgs);
+        _assertEquals(value1, testClass.getFieldValue(checkupInstance, attributeName), String.format(wrongValueMessage, className, attributeName));
+    }
+
+    @ParameterizedTest
+    @MethodSource("inputProvider")
+    public void lightbulbClassHasCorrectSetVoltageMethod(String value1, int value2) throws Throwable {
```

Comment:
I would avoid varying the string you use to initialize the object here. We are only concerned about one method, so we don't want to cause the test to fail based on something else entirely.

Reply:
You should also test changing the parameter to 0.